src/Aspire.Hosting.Azure.ContainerApps/AzureContanierAppsInfrastructure.cs

```diff
@@ -0,0 +1,991 @@
+
+                        // Special case references to keyvault secrets
+                        if (expr.Format == "{0}" && expr.ValueProviders.Count == 1 && secret == SecretType.KeyVault)
+                        {
+                            return (val, secret);
+                        }
+
+                        if (secret != SecretType.None)
+                        {
+                            finalSecretType = SecretType.Normal;
+                        }
+
+                        args[index++] = val;
+                    }
+
+                    return (new BicepValueFormattableString(expr.Format, args), finalSecretType);
+
+                }
+
+                throw new NotSupportedException("Unsupported value type " + value.GetType());
+            }
+
+            private BicepParameter AllocateVolumeStorageAccount(ContainerMountType type, string volumeIndex)
+            {
+                return AllocateParameter(VolumeStorageExpression.GetVolumeStorage(resource, type, volumeIndex));
+            }
+
+            private BicepParameter AllocateContainerImageParameter()
+                => AllocateParameter(ProjectResourceExpression.GetContainerImageExpression((ProjectResource)resource));
+
+            private BicepValue<int> AllocateTargetPortParameter()
+                => AllocateParameter(ProjectResourceExpression.GetTargetPortExpression((ProjectResource)resource));
+
+            private BicepParameter AllocateManagedIdentityIdParameter()
+                => _managedIdentityIdParameter ??= AllocateParameter(_containerAppEnviromentContext.ManagedIdentityId);
+
+            private void AllocateContainerRegistryParameters()
+            {
+                _containerRegistryUrlParameter ??= AllocateParameter(_containerAppEnviromentContext.ContainerRegistryUrl);
+                _containerRegistryManagedIdentityIdParameter ??= AllocateParameter(_containerAppEnviromentContext.ContainerRegistryManagedIdentityId);
+            }
+
+            private BicepParameter AllocateParameter(IManifestExpressionProvider parameter, Type? type = null)
+            {
+                if (!_allocatedParameters.TryGetValue(parameter, out var parameterName))
+                {
+                    parameterName = parameter.ValueExpression.Replace("{", "").Replace("}", "").Replace(".", "_").Replace("-", "_").ToLowerInvariant();
+
+                    if (parameterName[0] == '_')
+                    {
+                        parameterName = parameterName[1..];
+                    }
+
+                    _allocatedParameters[parameter] = parameterName;
+                }
+
+                if (!_bicepParameters.TryGetValue(parameterName, out var bicepParameter))
+                {
+                    var isSecure = parameter is BicepSecretOutputReference || parameter is ParameterResource { Secret: true };
```

Comment:
Would this make sense to have a helper method for this? So we didn't need to encode this logic in multple places?

Reply:
Not worth it. Actually, this is ineffective and the other piece of code is the one that matters. I'm going to need to resolve that problem in general though.